src/MessageList/MessageListBox.vala

```diff
@@ -39,6 +39,23 @@ public class Mail.MessageListBox : Gtk.ListBox {
         if (node.child != null) {
             go_down ((Camel.FolderThreadNode?) node.child);
         }
+
+        var children = get_children ();
+
+        if (children.length () == 1) {
+            children.foreach ((child) => {
```

Comment:
Not a huge fan of using a foreach for a single element list. Can you use `nth_data` or `get_row_at_index` again instead?